.github/workflows/full_stack_tests.yml

```diff
@@ -522,3 +522,34 @@ jobs:
       - name: Fail if workflow status is failure
         if: ${{ steps.check_workflow_status.outputs.WORKFLOW_STATUS == 'failure' }}
         run: exit 1
+  trigger_build_and_deploy:
+    name: Trigger build and deploy
+    needs: [check_workflow_statuses]
+    runs-on: ubuntu-22.04
+    if: ${{ github.event_name == 'push' && github.repository == 'oppia/oppia' && github.ref == 'refs/heads/develop'}}
```

Comment:
This should ensure that the "build and deploy" only happens for commits on the develop branch on the oppia/oppia repo.

Reply:
This is interesting ... are we OK with build+deploy happening if the full-stack tests pass, but possibly if the other tests (like unit tests etc.) fail?

I'm actually leaning slightly towards "yeah it's ok", just wanted to note that this workflow doesn't guarantee all checks outside this workflow pass. WDYT?

Reply:
I think it should be fine.

If we do plan to update this in the future, we could convert the existing YAML files for unit tests, full_stack_tests and others into [reusable workflows](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow) and have a "master" workflows that will execute the individual test workflows and finally trigger the build+deploy.